rm-datasource/src/main/java/io/seata/rm/datasource/ConnectionProxy.java

```diff
@@ -211,24 +211,30 @@ private void processLocalCommitWithGlobalLocks() throws SQLException {
     }
 
     private void processGlobalTransactionCommit() throws SQLException {
-        try {
-            register();
-        } catch (TransactionException e) {
-            recognizeLockKeyConflictException(e, context.buildLockKeys());
-        }
-
-        try {
-            if (context.hasUndoLog()) {
+        if (context.hasUndoLog()) {
+            try {
+                register();
```

Comment:
I think it's better to put it in registers and flushUndoLogs.
if there is no undolog why flushUndoLogs method execution without judgment

Reply:
I have optimized the code, PTAL

Reply:
I have finish it.PTAL